schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/ISchemaRegistryService.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2016 Hortonworks.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.registries.schemaregistry;
+
+import com.hortonworks.registries.schemaregistry.errors.IncompatibleSchemaException;
+import com.hortonworks.registries.schemaregistry.errors.InvalidSchemaException;
+import com.hortonworks.registries.schemaregistry.errors.SchemaNotFoundException;
+import com.hortonworks.registries.schemaregistry.serde.SerDesException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+
+/**
+ * Basic service interface for schema registry which should be implemented by client and server interfaces.
+ *
+ */
+public interface ISchemaRegistryService {
+
+    /**
+     * @return Collection of supported schema providers. For ex: avro.
+     */
+    Collection<SchemaProviderInfo> getSupportedSchemaProviders();
+
+    /**
+     * Registers information about a schema.
+     *
+     * @param schemaMetadata metadata about schema.
+     * @return id of the registered schema which is successfully registered now or earlier.
+     */
+    Long registerSchemaMetadata(SchemaMetadata schemaMetadata);
```

Comment:
for the small breaking change should we not rename this as add, or rename the adds as register (seems inconsistent)